lib/rules/no-required-prop-with-default.js

```diff
@@ -0,0 +1,154 @@
+/**
+ * @author @neferqiqi
+ * See LICENSE file in root directory for full license.
+ */
+'use strict'
+// ------------------------------------------------------------------------------
+// Requirements
+// ------------------------------------------------------------------------------
+
+const utils = require('../utils')
+/**
+ * @typedef {import('../utils').ComponentTypeProp} ComponentTypeProp
+ */
+
+// ------------------------------------------------------------------------------
+// Helpers
+// ------------------------------------------------------------------------------
+
+// ...
+
+// ------------------------------------------------------------------------------
+// Rule Definition
+// ------------------------------------------------------------------------------
+
+module.exports = {
+  meta: {
+    type: 'suggestion',
```

Comment:
I think this should be an error. @ota-meshi what do you think?

```suggestion
    type: 'error',
```

Reply:
I agree with you 👍

Reply:
Eslint prompts "`meta.type` must be either `problem`, `suggestion`, or `layout`"; do you mean this should be 'problem'?

Reply:
Ah, yeah, I meant `problem`.

Reply:
done